tests/Subscriber/GitRepoSubscriberTest.php

```diff
@@ -11,67 +11,341 @@
+        $this->assertEquals('gushphp', $command->getIssueTracker()->getUsername());
+        $this->assertEquals('gush', $command->getIssueTracker()->getRepository());
+    }
+
+    public function testAllowsToOverwriteConfiguredOrgAndRepoForIssue()
+    {
+        $command = new GitRepoCommand();
+        $commandTest = $this->runCommandTest(
+            $command,
+            ['--repo-adapter' => 'github_enterprise', '--issue-org' => 'cordoval', '--issue-project' => 'gush-sandbox'],
+            ['repo_adapter' => 'github', 'repo_org' => 'gushphp', 'repo_repo' => 'gush']
+        );
+
+        $display = $commandTest->getDisplay(true);
+
+        $this->assertNotContains('You did not set or provide an adapter-name', $display);
+
+        $this->assertNotNull($command->getAdapter());
+        $this->assertEquals('github_enterprise', $command->getAdapter()->getAdapterName());
+
+        $this->assertEquals('gushphp', $command->getAdapter()->getUsername());
+        $this->assertEquals('gush', $command->getAdapter()->getRepository());
+
+        $this->assertEquals('cordoval', $command->getIssueTracker()->getUsername());
+        $this->assertEquals('gush-sandbox', $command->getIssueTracker()->getRepository());
+    }
+
+    public function testThrowsErrorOnInvalidRepoAdapter()
+    {
+        $this->setExpectedException(
+            UserException::class,
+            'Adapter "noop-noop" (for repository-management) does not seem to be supported'
+        );
+
+        $command = new GitRepoCommand();
+        $this->runCommandTest(
+            $command,
+            ['--repo-adapter' => 'noop-noop'],
+            ['repo_adapter' => 'github']
+        );
+    }
+
+    public function testThrowsErrorOnInvalidIssueAdapter()
+    {
+        $this->setExpectedException(
+            UserException::class,
+            'Adapter "noop-noop" (for issue-tracking) does not seem to be supported'
+        );
+
+        $command = new GitRepoCommand();
+        $this->runCommandTest(
+            $command,
+            ['--issue-adapter' => 'noop-noop'],
+            ['repo_adapter' => 'github']
+        );
+    }
+
+    public function testThrowsErrorWhenNotConfiguredAndNoRemoteIsSetForAutoDetection()
+    {
+        $this->setExpectedException(
+            UserException::class,
+            'Unable to get the repository information, Git remote "origin" should be set for automatic detection'
+        );
+
+        $command = new GitRepoCommand();
+        $this->runCommandTest(
+            $command,
+            [],
+            [],
+            function (HelperSet $helperSet) {
+                $helperSet->set($this->getGitConfigHelper(false)->reveal());
+            }
+        );
+    }
+
+    public function testThrowsErrorWhenNoOptionsGivenAndNotInGitDir()
+    {
+        $this->setExpectedException(
+            UserException::class,
+            'Provide the --org and --repo options when your are outside of a Git directory.'
```

Comment:
No, you to provide them (or set/supply) not sure which is the correct spelling :)